### PR TITLE
[AutoParallel-PIR] Basic Pipeline Parallelism 

### PR DESCRIPTION
**Pull request by JZ-LIANG**
### PR Category

Auto Parallel

### PR Types

New features 

### Description

Pcard-67164

Update the logic of:

- Reshard Op
- Rehsard Pass
- Partition Pass

to support a baisc pipeline parallelism in AutoParallel PIR mode

### CODE REVIEW

File: paddle/fluid/pir/dialect/distributed/ir/dist_op.cc [hunk 1]
```diff
@@ -271,11 +271,37 @@ void ReshardOp::VerifySig() {
...
+      merged_ids,
+      {"merged"});
+}
+
 void ReshardOp::Build(pir::Builder& builder,
                       pir::OperationArgument& argument,
                       pir::Value input,
                       TensorDistAttribute tensor_dist_attr) {
-  VLOG(4) << "Start build ReshardOp";
+  VLOG(0) << "Start build ReshardOp";
```
Comment (winter-wang) [inline paddle/fluid/pir/dialect/distributed/ir/dist_op.cc]: 这儿调为0是不是有点高了？

File: test/auto_parallel/pir/mlp_demo.py [hunk 2]
```diff
@@ -14,76 +14,192 @@
...
-        dist_model._engine._initialize(mode)
-        dist_model._engine._executor = paddle.static.Executor(
-            _current_expected_place()
+        dist_loader = dist.shard_dataloader(loader, meshes=[mesh1, mesh2])
+        dist_model = dist.to_static(pp_layer, dist_loader, loss_fn, opt)
+
+        engine = dist_model._engine
+        engine._build("train")
+        dist_program = engine._fwd_main_progs["train"]
+        print("================" + "fwd mix program" + "================")
```
Comment (winter-wang) [inline test/auto_parallel/pir/mlp_demo.py]: 当前python单测里面似乎不允许用print打印信息。

Comment (JZ-LIANG) [reply test/auto_parallel/pir/mlp_demo.py]: 纯debug，后续删除

File: paddle/fluid/pir/dialect/distributed/ir/dist_op.cc (hunk 1 above)
Comment (JZ-LIANG) [reply paddle/fluid/pir/dialect/distributed/ir/dist_op.cc]: 纯debug，后续删除

**Review by winter-wang — APPROVED** [review]

**Pull request merged**